webapp/src/components/surveyPost/surveyPost.tsx

```diff
@@ -0,0 +1,243 @@
+
+        const responsesExist = survey.response !== undefined;
+        const surveyExpired = survey.status === 'ended';
+
+        setDisabled(responsesExist || surveyExpired);
+        setExpired(surveyExpired);
+    };
+
+    // fetch data on initial mount
+    useEffect(() => {
+        fetchUserSurvey(post.props.surveyID);
+    }, [post.props]);
+
+    const questionResponseChangeHandler = useDebouncedCallback(
+        (questionID: string, response: unknown) => {
+            const newDraftResponse: SurveyResponse = draftResponse.current || {
+                responses: {},
+                dateCreated: format(new Date(), 'dd/MM/yyyy'),
+            };
+            newDraftResponse.responses[questionID] = response;
+            draftResponse.current = newDraftResponse;
+
+            // if this is the system rating question, submit response ASAP
+            if (questionID === linearScaleQuestionID.current) {
+                submitRating();
+            }
+        },
+        500,
+    );
+
+    const submitSurveyHandler = useCallback(async () => {
+        if (!draftResponse.current ||
+            Object.keys(draftResponse.current?.responses).length === 0
+        ) {
+            return;
+        }
+
+        const response = await submitSurveyResponse();
+        if (response.success && survey !== undefined) {
+            const newSurvey: UserSurvey = {
+                ...survey,
+                response: draftResponse.current,
+            };
+            setSurvey(newSurvey);
+            setErrorMessage('');
+        } else if (response.error) {
+            setErrorMessage('Failed to submit survey response. Please try again.');
+        }
+    }, [survey]);
+
+    const submitSurveyResponse = async () => {
+        // make API call here. Send draftResponse.current as payload...
+        return {success: true, error: false};
+    };
+
+    // this function is to submit the linear scale rating as soon as a user selects it,
+    // even without pressing the submit button.
+    const submitRating = async () => {
+        if (!linearScaleQuestionID.current || !draftResponse.current) {
+            return;
+        }
+
+        const ratingQuestionResponse = draftResponse.current?.responses[linearScaleQuestionID.current];
+        // eslint-disable-next-line @typescript-eslint/no-unused-vars
+        const payload: SurveyResponse = {
+            responses: {
+                [linearScaleQuestionID.current]: ratingQuestionResponse,
+            },
+            dateCreated: draftResponse.current?.dateCreated,
+        };
+
+        // send payload to submit survey response API here...
+    };
+
+    const renderedMessage = useMemo(() => {
+        // @ts-expect-error window is definitely MattermostWindow in plugins
+        const mmWindow = window as MattermostWindow;
+        const htmlString = mmWindow.PostUtils.formatText(post.message, {markdown: true});
+        return mmWindow.PostUtils.messageHtmlToComponent(htmlString, isRHS);
```

Comment:
`isRHS` is actually no longer used, so you're safe to remove it

Reply:
Done.